pkg/scheduler/plugins/util/util.go

```diff
@@ -272,29 +274,32 @@ func (nl *NodeLister) List() ([]*v1.Node, error) {
 }
 
 // NormalizeScore normalizes the score for each filteredNode
-func NormalizeScore(maxPriority int64, reverse bool, scores map[string]int64) {
-	var maxCount int64
-	for _, score := range scores {
-		if score > maxCount {
-			maxCount = score
+func NormalizeScore(maxPriority int64, reverse bool, scores []api.ScoredNode) {
+	var (
+		maxCount int64 = math.MinInt64
+		minCount int64 = math.MaxInt64
+	)
+
+	for _, scoreNode := range scores {
+		if scoreNode.Score > maxCount {
+			maxCount = scoreNode.Score
+		}
+		if scoreNode.Score < minCount {
+			minCount = scoreNode.Score
 		}
 	}
 
-	if maxCount == 0 {
-		if reverse {
-			for key := range scores {
-				scores[key] = maxPriority
-			}
+	maxMinDiff := maxCount - minCount
+	for idx := range scores {
+		tmpScore := int64(0)
+
+		if maxMinDiff > 0 {
+			tmpScore = maxPriority * (scores[idx].Score - minCount) / maxMinDiff
```

Comment:
Can you clarify why we need to change the calculation way here?

Reply:
l'm ok with both, it seems that both of the calculation methods are used in k8s )

Reply:
Yes, So it's better to keep the current normalization.